userapi/storage/pushers/sqlite3/pushers_table.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2021 Dan Peleg <dan@globekeeper.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlite3
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/matrix-org/dendrite/internal/sqlutil"
+	"github.com/matrix-org/dendrite/userapi/api"
+	"github.com/sirupsen/logrus"
+
+	"github.com/matrix-org/dendrite/clientapi/userutil"
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
+const pushersSchema = `
+-- Stores data about pushers.
+CREATE TABLE IF NOT EXISTS pusher_pushers (
+	localpart TEXT,
+	session_id BIGINT,
+	profile_tag TEXT,
+	kind TEXT,
+	app_id VARCHAR(64),
+	app_display_name TEXT,
+	device_display_name TEXT,
+	pushkey VARCHAR(512),
+	lang TEXT,
+	data TEXT,
+
+	UNIQUE (app_id, pushkey, localpart)
```

Comment:
If this is the unique key then I'm surprised to see selecting pushers and updating pushers use `WHERE localpart = $1 AND pushkey = $2`.